packages/web-components/fast-foundation/src/number-field/number-field.ts

```diff
@@ -119,13 +119,22 @@ export class NumberField extends FormAssociatedNumberField {
      */
     @attr({ converter: nullableNumberConverter })
     public max: number;
+
+    /**
+     * Ensures that the max is greater than the min and that the value
+     *  is less than the max
+     * @param previousValue - the previous max value
+     * @param nextValue - updated max value
+     *
+     * @internal
+     */
     public maxChanged(previousValue, nextValue): void {
         this.max = Math.max(nextValue, this.min ?? nextValue);
         const min = Math.min(this.min, this.max);
         if (this.min !== undefined && this.min !== min) {
             this.min = min;
         }
-        this.valueChanged(this.value, this.value);
+        this.valueChanged(null, this.value);
```

Comment:
I'm not a fan of this since `prevValue` _isn't_ null, it's the value of `this.value`. I think the contents of `valueChanged` should be extracted into their own clamping function, which can be called by `minChanged`, `maxChanged`, and `valueChanged` on an independent basis.

Reply:
I was previously calling `this.valueChanged(this.value, this.value)` because it was technically accurate.

The big problem is validating and changing this.value from inside this.valueChanged without re-triggering this.valueChanged.

I will abstract the validation and call into it from minChanged, maxChanged and valueChanged.

Reply:
I've pull out the validation code to it's own function `getValidValue()` to make sure the value can be parsed into a valid number and is between the min and max. `valueChanged()` calls into it as does `minChanged()` and `maxChanged()`. The big difference here is that typing in the text field now passes a flag to `valueChanged()` telling it to not override the contents of `this.control.value`.